### PR TITLE
added insertion site code and tests

### PR DESCRIPTION
**Pull request by jmmshn**
## Added new class for general site-insertion algorithm

- Added ChargeInsertionAnalyzer class and tests
- Reformatted and ignoring .pre-commit-config.yaml

### CODE REVIEW

File: pymatgen/analysis/defects/core.py [hunk 1]
```diff
@@ -17,7 +17,6 @@
 from pymatgen.core.structure import Structure, PeriodicSite
 from pymatgen.core.composition import Composition
 from pymatgen.symmetry.analyzer import SpacegroupAnalyzer
-from pymatgen.analysis.defects.utils import kb
```
Comment (jmmshn) [inline pymatgen/analysis/defects/core.py:20]: This had to be moved due to circular imports

File: pymatgen/analysis/defects/utils.py [hunk 2]
```diff
@@ -1279,28 +1324,144 @@ def sort_sites_by_integrated_chg(self, r=0.4):
...
+        AA, BB, CC = np.meshgrid(aa, bb, cc, indexing="ij")
         dist_from_pos = self.chgcar.structure.lattice.get_all_distances(
             fcoords1=np.vstack([AA.flatten(), BB.flatten(), CC.flatten()]).T,
-            fcoords2=pos_frac)
+            fcoords2=pos_frac,
+        )
         return dist_from_pos.reshape(AA.shape)
 
 
+class ChargeInsertionAnalyzer(ChargeDensityAnalyzer):
```
Comment (jmmshn) [inline pymatgen/analysis/defects/utils.py:1351]: New class is here

File: pymatgen/analysis/defects/tests/test_utils.py [hunk 3]
```diff
@@ -367,14 +428,39 @@ def test_sort_sites_by_integrated_chg(self):
...
+        self.assertAlmostEqual(ca._extrema_df.iloc[0]["Charge Density"], 1.65288944124)
+        self.assertAlmostEqual(
+            ca._extrema_df.iloc[0]["avg_charge_den"], 0.006831484178753711
+        )
+
+
+@unittest.skipIf(
+    not peak_local_max, "skimage.feature.peak_local_max module not present."
+)
+class TestChargeInsertionAnalyzer(unittest.TestCase):
```
Comment (jmmshn) [inline pymatgen/analysis/defects/tests/test_utils.py:444]: Tests here.

File: pymatgen/analysis/defects/core.py [hunk 4]
```diff
@@ -29,13 +28,15 @@
 
 logger = logging.getLogger(__name__)
 
+kb = 8.6173324e-5  # eV / K
```
Comment (mkhorton) [inline pymatgen/analysis/defects/core.py:31]: Add this to `pymatgen.core.units` if it's not already there and import from there.

Comment (jmmshn) [reply pymatgen/analysis/defects/core.py]: Done, the pylint caught some problems with the defect utils code.  Should be fixed now.

Comment (jmmshn) [reply pymatgen/analysis/defects/core.py]: Should we have a pre-commit-config.yaml for pymatgen?  I think currently the precommit only does pycodestyle

Comment (mkhorton) [reply pymatgen/analysis/defects/core.py]: Sure, as long as we have some description in the docs to explain what it does and how to use it. For people who do not have pre-commit installed it's ignored correct?

Comment (jmmshn) [reply pymatgen/analysis/defects/core.py]: > For people who do not have pre-commit installed it's ignored correct?
Correct

**Comment by mkhorton** [comment]
Thanks @jmmshn, going to merge this.

**Pull request merged**